packages/preview/anatomy/0.1.0/lib.typ

```diff
@@ -0,0 +1,77 @@
+#let metrics(
+  size, typeface,
+  _text,
+  top-edges: ( "ascender", "cap-height", "x-height", "baseline" ),
+  bottom-edges: ( "descender", )
+) = {
+  set text(
+    size,
+    font: typeface,
+  )
+
+  let edges = top-edges + bottom-edges
+
+  // Calulate metrics and update their corresponding states
+  style(styles => {
+    let set-metric(edge, above-baseline: true) = {
+      let frame = measure(text(
+        top-edge: if above-baseline { edge } else { "baseline" },
+        bottom-edge: if above-baseline { "baseline" } else { edge }
+      )[.], styles)
+
+      state(edge).update(
+        (if above-baseline { 1 } else { -1 } * frame.height)
+      )
+    }
+
+    for edge in top-edges { set-metric(edge) }
+    for edge in bottom-edges { set-metric(edge, above-baseline: false) }
```

Comment:
It's not really any of my business, but why does this need `state`? It seems like the measurement could just happen in a `locate(loc => style(styles => { ...` without any need for state.

Reply:
Thank you for your advice. I was thinking about the values of metrics (`ascender` height, etc.) can be used somewhere else in future versions, and exposed to users. I wonder whether there is a simpler way to store global values without using `state`.

Reply:
A function like locate/style/.. that takes a closure and provides the measured metrics would be more idiomatic I'd say.

Reply:
I have removed `state`s (using `style(styles => locate(loc => { ... })`. Does the order matter?).

I used closure for additional typesetting. I’m afraid that I couldn’t find documentation of how to pass a closure as parameter, so it might not be the standard way.

Thank you for your time.

Reply:
The order doesn't really matter. It also wasn't my intention to nitpick your package implementation, I was just curious.